### PR TITLE
Add completed execution on to submission

### PR DESCRIPTION
**Pull request by msmner**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/825

Linked pull requests from another repos (if any):

**Summary of the changes made**:
1.  Add property CompletedExecutionOn to Submissions table

### CODE REVIEW

File: Services/Worker/OJS.Services.Worker.Business/Implementations/SubmissionsBusinessService.cs [hunk 1]
```diff
@@ -148,7 +148,9 @@ private static void PreprocessSubmission(SubmissionServiceModel submission)
         var ojsWorkerExecutionResult = this.submissionExecutor.Execute<TInput, TResult>(ojsSubmission);
 
         var executionResult = this.mapper.Map<ExecutionResultServiceModel>(ojsWorkerExecutionResult);
+
         executionResult.StartedExecutionOn = ojsSubmission.StartedExecutionOn;
+        executionResult.CompletedExecutionOn = DateTime.UtcNow;
```
Comment (gogo4ds) [inline Services/Worker/OJS.Services.Worker.Business/Implementations/SubmissionsBusinessService.cs:153]: Lets move CompletedExecutionOn in the end of the execution, just before returning the result and make sure StartedExecutionOn is at the beggining.

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Services/Worker/OJS.Services.Worker.Business/Implementations/SubmissionsBusinessService.cs (hunk 1 above)
Comment (msmner) [reply Services/Worker/OJS.Services.Worker.Business/Implementations/SubmissionsBusinessService.cs:153]: Done.

File: Services/Worker/OJS.Services.Worker.Business/ExecutionContext/Implementations/ExecutionContextBuilderService.cs [hunk 2]
```diff
@@ -50,6 +50,8 @@ public OjsSubmission<TInput> BuildOjsSubmission<TInput>(SubmissionServiceModel s
                 string.Format(CannotCreateInputTemplate, submissionServiceModel.ExecutionType))
             .VerifyResult();
 
+        submission.StartedExecutionOn = DateTime.UtcNow;
```
Comment (gogo4ds) [inline Services/Worker/OJS.Services.Worker.Business/ExecutionContext/Implementations/ExecutionContextBuilderService.cs:53]: `BuildOjsSubmission` should be included in the calculation. It is part of the execution. Move as close as possible to the entry point. Ideally when the request enters.

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Services/Worker/OJS.Services.Worker.Business/ExecutionContext/Implementations/ExecutionContextBuilderService.cs (hunk 2 above)
Comment (msmner) [reply Services/Worker/OJS.Services.Worker.Business/ExecutionContext/Implementations/ExecutionContextBuilderService.cs]: Done.

Comment (gogo4ds) [reply Services/Worker/OJS.Services.Worker.Business/ExecutionContext/Implementations/ExecutionContextBuilderService.cs]: Can you put it before the validation logic also? Should be the first thing in the service

Comment (msmner) [reply Services/Worker/OJS.Services.Worker.Business/ExecutionContext/Implementations/ExecutionContextBuilderService.cs]: Done.

File: Services/Worker/OJS.Services.Worker.Business/ExecutionContext/Implementations/ExecutionContextBuilderService.cs [hunk 3]
```diff
@@ -43,15 +43,15 @@ public OjsSubmission<TInput> BuildOjsSubmission<TInput>(SubmissionServiceModel s
     {
         var submission = submissionServiceModel.Map<OjsSubmission<TInput>>();
 
+        submission.StartedExecutionOn = submissionServiceModel.StartedExecutionOn;
```
Comment (gogo4ds) [inline Services/Worker/OJS.Services.Worker.Business/ExecutionContext/Implementations/ExecutionContextBuilderService.cs:46]: Moving this does absolutely nothing to the end result. This is just copying the value from one model, to another.

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Services/Worker/OJS.Services.Worker.Business/ExecutionContext/Implementations/ExecutionContextBuilderService.cs (hunk 2 above)
Comment (gogo4ds) [reply Services/Worker/OJS.Services.Worker.Business/ExecutionContext/Implementations/ExecutionContextBuilderService.cs]: I men to move the point where StartedExecutionOn is assigned the DateTime.UtcNow value. We want to start measuring this time as soon as the submission hits the worker.

Comment (msmner) [reply Services/Worker/OJS.Services.Worker.Business/ExecutionContext/Implementations/ExecutionContextBuilderService.cs]: Done.

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**